src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelLockFileModule.java

```diff
@@ -107,15 +107,19 @@ public void afterCommand() throws AbruptExitException {
         updatedExtensionMap = new HashMap<>();
 
     // Keep old extensions if they are still valid.
-    ImmutableTable<ModuleExtensionId, ModuleKey, ModuleExtensionUsage> oldExtensionUsages =
-        BazelDepGraphFunction.getExtensionUsagesById(oldLockfile.getModuleDepGraph());
     for (var entry : oldLockfile.getModuleExtensions().entrySet()) {
       var moduleExtensionId = entry.getKey();
-      var factorToLockedExtension = entry.getValue();
-      ModuleExtensionEvalFactors firstEntryFactors =
-          factorToLockedExtension.keySet().iterator().next();
-      if (shouldKeepExtension(moduleExtensionId, firstEntryFactors, oldExtensionUsages)) {
-        updatedExtensionMap.put(moduleExtensionId, factorToLockedExtension);
+      ImmutableMap.Builder<ModuleExtensionEvalFactors, LockFileModuleExtension>
+          factorsToKeepBuilder = new ImmutableMap.Builder<>();
+      for (var factorAndExtension : entry.getValue().entrySet()) {
```

Comment:
this now checks every "factor set", instead of just the first one. I think just checking the first one is slightly simpler? (plus it avoids duplicate work -- all "factor sets" should chave the same usage hash anyway, right?)